python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -387,6 +391,39 @@ def get_asset_keys_to_resource_props(
     }
 
 
+def get_updated_cli_invocation_params_for_context(
+    context: Union[OpExecutionContext, AssetExecutionContext],
+    manifest: Mapping[str, Any],
+    dagster_dbt_translator: "DagsterDbtTranslator",
+):
+    assets_def: Optional[AssetsDefinition] = None
+    with suppress(DagsterInvalidPropertyError):
+        assets_def = context.assets_def if context else assets_def
+
+    selection_args: list[str] = []
+    indirect_selection = os.getenv(DBT_INDIRECT_SELECTION_ENV, None)
+    if context and assets_def is not None:
```

Comment:
would context even be None here? if so, should represent that in the type signature, otherwise, can remove this part of the statement for clarity.

Reply:
The context can be None, when the cli invocation is made outside an asset or an op - signature updated in [4a9ac16](https://github.com/dagster-io/dagster/pull/29650/commits/4a9ac16a113b92be5d856f24c4867e8d97c70f5d)